src/views/Graph.vue

```diff
@@ -0,0 +1,625 @@
+              action: 'callback',
+              callback: this.reset
+            },
+            {
+              title: 'Increase Spacing',
+              icon: mdiArrowExpand,
+              action: 'callback',
+              callback: this.increaseSpacing
+            },
+            {
+              title: 'Decrease Spacing',
+              icon: mdiArrowCollapse,
+              action: 'callback',
+              callback: this.decreaseSpacing
+            }
+          ]
+        }
+      ]
+    }
+  },
+  mounted () {
+    // allow render to happen before we go configuring svgPanZoom
+    const self = this
+    this.$nextTick(function () {
+      // NOTE: Initiating svgPanZoom may result in some "violation" warnings
+      // see https://github.com/bumbu/svg-pan-zoom/issues/408
+      self.panZoomWidget = svgPanZoom(
+        self.$refs.graph,
+        {
+          // NOTE: fix must be false otherwise it's trying to measure up before
+          // the viewport is loaded or something like that which causes
+          // NaN values to end up in the transformation matrix
+          // TODO: enable the "thumbnail" viewer (i.e. minimap, see svg-pan-zoom)
+          viewportSelector: '.svg-pan-zoom_viewport',
+          panEnabled: true,
+          controlIconsEnabled: false,
+          zoomEnabled: true,
+          dblClickZoomEnabled: true,
+          mouseWheelZoomEnabled: true,
+          preventMouseEventsDefault: true,
+          zoomScaleSensitivity: 0.2,
+          minZoom: 0.01, // how zoomed out we can go
+          maxZoom: 10, // how zoomed in we can go
+          fit: false,
+          contain: false,
+          center: true,
+          refreshRate: 'auto'
+        }
+      )
+      self.updateTimer()
+    })
+  },
+  beforeDestroy () {
+    clearInterval(this.refreshTimer)
+  },
+  computed: {
+    ...mapState('workflows', ['cylcTree']),
+    ...mapGetters('workflows', ['getNodes']),
+    query () {
+      return new SubscriptionQuery(
+        QUERY,
+        this.variables,
+        'workflow',
+        [
+          // TODO: this callback should be run automatically
+          // (only one instance for all views)
+          // new CylcTreeCallback()
+        ]
+      )
+    },
+    workflowIDs () {
+      return [this.workflowId]
+    },
+    workflows () {
+      return this.getNodes('workflow', this.workflowIDs)
+    }
+  },
+  methods: {
+    setOption (option, value) {
+      Vue.set(this, option, value)
+    },
+    updateTimer () {
+      // turn the timer on or off depending on the value of autoRefresh
+      if (this.autoRefresh) {
+        this.refreshTimer = setInterval(this.refresh, 2000)
+      } else {
+        clearInterval(this.refreshTimer)
+        this.refreshTimer = null
+      }
+    },
+    increaseSpacing () {
+      // increase graph layout node spacing by 10%
+      this.spacing = this.spacing * 1.1
+    },
+    decreaseSpacing () {
+      // decrease graph layout node spacing by 10%
```

Comment:
This is reducing spacing by 10% of the final value, so that if you increase the final value by 10% you get back to where you started. I wonder if it might be worth pointing this. If it isn't web-boilerplate obvious to everyone who's done this sort of thing before.

Reply:
It's pretty irrelevant, I just like the numerical niceness of getting back to where we started.